### PR TITLE
ENH: Add scraping of other IDs

### LINKED ISSUE #108: Scrape all ID types from paper collection
As a plugin user,
I would like the `collection-scraper` to  output accession IDs of all types (incl. study and experiments IDs) independent on whether they are covered by `get-sequences`, `get-metadata` and `get-all`. Such that as a user I know which datasets were linked with my paper collection.

### PR DESCRIPTION
**Pull request by adamovanja**
closes #108

This PR adds scraping of study, sample and experiment IDs to the `scrape-collection` action. As sample and experiment IDs are not supported by the other q2fondue actions, they are outputted into one common artifact `other-ids.qza`.

**Testing**
Run the `scrape-collection` action on a Zotero collection of your own. Or add any of the below publications to a collection and verify that the study and "other" IDs were scraped correctly:

* Publications with study IDs: 10.1016/j.cell.2014.09.053, 10.1038/s41587-019-0209-9
* Publication with experiment IDs: 10.1038/s41467-019-13036-1

### CODE REVIEW

File: tutorial/tutorial.md [hunk 1]
```diff
@@ -215,17 +215,19 @@ qiime fondue get-sequences \
...
 ```shell
 qiime fondue scrape-collection \
               --p-library-type user \
               --p-user-id user_id \
               --p-api-key my_key \
               --p-collection-name collection_name \
               --o-run-ids fondue-output/run_ids.qza \
+              --o-study-ids fondue-output/study_ids.qza \
               --o-bioproject-ids fondue-output/bioproject_ids.qza \
+              --o-other-ids other_ids.qza \
```
Comment (misialq) [inline tutorial/tutorial.md:230]:
```suggestion
              --o-other-ids fondue-output/other_ids.qza \
```

**Review by lina-kim — COMMENTED** [review]
Runs successfully with other IDs, thanks for providing examples. 

A limiting issue is that all of these search queries will extract IDs as-is and won't interpret + expand text e.g. `SRX1479846–50` from your [10.1038/s41467-019-13036-1 example](https://www.nature.com/articles/s41467-019-13036-1). Even worse, I realized the character used as a separator ("–") isn't even a canonical hyphen :(

Also, I'm not sure if this is limited to my system/if `metadata tabulate` just hates me; but I get this issue when I try to generate a visualization instead of `qiime tools extract`:
```
(qiime2-2022.2) $ qiime metadata tabulate --m-input-file other_ids.qza --o-visualization other_ids.qzv

There was an issue with loading the file other_ids.qza as metadata:

  Metadata file must be encoded as UTF-8 or ASCII. The following error occurred when decoding the file:

  'utf-8' codec can't decode byte 0xb0 in position 12: invalid start byte
```

**Comment by adamovanja** [comment]
Thank you both for the reviews.

@misialq:

1. I believe we can view the `NCBIAccessionIDs` type as IDs that uniquely identify particular records in NCBI databases, which both study and experiment IDs do in the SRA database. Hence, I do not see a problem here. 
2. I have encountered publications that only report experiment IDs instead of the associated study and/or BioProject IDs. Given that best practices in reporting accession IDs are not enforced, I think adding a support for these other IDs is a benefit for the user trying to retrieve all raw data used in a particular set of studies. 
3. Yes, I am planning to add support for these IDs too. As for the need of scraping these IDs: when one starts off by scraping a library, it is not given that the associated run/study/BioProject ID is provided in the publication text (see also my answer in 2).

Let me know what you think and thanks for the inline comments. 

@lina-kim thanks for spotting the "hyphen" issue. I actually added some special cases where IDs are not just fetched as-is (see here: https://github.com/bokulich-lab/q2-fondue/blob/0e1113661583e1f5cd948155526c7680a0538cdb/q2_fondue/scraper.py#L242-L256). But the scraper currently does not support hyphens yet. I will definitely add this in the future. 
As for the `metadata tabulate` issue: Is this something you only encounter with q2-fondue outputs or with any other Q2 plugin? I personally do not have this error occurring with q2-fondue outputs or any other ones.

File: q2_fondue/scraper.py [hunk 2]
```diff
@@ -325,27 +339,25 @@ def scrape_collection(
...
+    for type in doi_dicts.keys():
+        if len(doi_dicts[type]) == 0:
+            logger.warning(f'The provided collection {collection_name} '
+                           f'does not contain any {type} IDs')
 
     dfs = []
-    for doi_dict in (run_doi, bioproject_doi):
+    for doi_dict in (
+            doi_dicts['run'], doi_dicts['study'], doi_dicts['bioproject'],
+            doi_dicts['other']):
```
[the remark this answers is not in the document]
Comment (adamovanja) [reply q2_fondue/scraper.py:360]: ah, I had missed that since Python 3.6 `dict` preserves orders 🙌🏼

**Comment by lina-kim** [comment]
> @lina-kim thanks for spotting the "hyphen" issue. I actually added some special cases where IDs are not just fetched as-is (see here:
> 
> https://github.com/bokulich-lab/q2-fondue/blob/0e1113661583e1f5cd948155526c7680a0538cdb/q2_fondue/scraper.py#L242-L256
> 
> ). But the scraper currently does not support hyphens yet. I will definitely add this in the future (see new issues #118).

Thanks for addressing those cases @adamovanja!

> As for the `metadata tabulate` issue: Is this something you only encounter with q2-fondue outputs or with any other Q2 plugin? I personally do not have this error occurring with q2-fondue outputs or any other ones.

Oh interesting. I don't have an issue with other QIIME outputs, but with fondue outputs I keep getting the following error when trying to run `qiime metadata tabulate`. It must be something on my end, I'll look into it.
```
There was an issue with loading the file metadata.qza as metadata:

  Metadata file must be encoded as UTF-8 or ASCII. The following error occurred when decoding the file:

  'utf-8' codec can't decode byte 0xb7 in position 17: invalid start byte
```

**Comment by adamovanja** [comment]
> Oh interesting. I don't have an issue with other QIIME outputs, but with fondue outputs I keep getting the following error when trying to run `qiime metadata tabulate`. It must be something on my end, I'll look into it.

@lina-kim if the error persists, it would be great if you could open an issue on this repos with the particular set of publications with which the scraped Metadata file returns this error. I'm happy to look into it.

**Review by misialq — APPROVED** [review]
Looks good! 🚀

**Pull request merged**